src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/TrackedEntityRelationshipsWrapper/hooks/useDataEntryReduxConverter.js

```diff
@@ -96,31 +98,68 @@ export const useDataEntryReduxConverter = ({
     const dataEntryFieldValues = useSelector(({ dataEntriesFieldsValue }) => dataEntriesFieldsValue[dataEntryKey]);
     const dataEntryFieldsMeta = useSelector(({ dataEntriesFieldsMeta }) => dataEntriesFieldsMeta[dataEntryKey]);
     const { programId, orgUnit } = useSelector(({ newRelationshipRegisterTei }) => newRelationshipRegisterTei);
+    const { formFoundation: scopeFormFoundation } = useMetadataForRegistrationForm({ selectedScopeId });
+    const { firstStageMetaData } = useBuildFirstStageRegistration(programId, !programId);
+    const { formFoundation } = useMergeFormFoundationsIfApplicable(scopeFormFoundation, firstStageMetaData);
 
-    const buildTeiPayload = () => {
-        const { form: formFoundation } = getMetadata(programId, trackedEntityTypeId);
+    const buildTeiWithEnrollment = () => {
+        if (!formFoundation) return null;
+        const firstStage = firstStageMetaData && firstStageMetaData.stage;
         const clientValues = getClientValuesForFormData(formValues, formFoundation);
-        const serverValuesForFormValues = formFoundation.convertValues(clientValues, convertClientToServer);
+        const serverValuesForFormValues = formFoundation.convertAndGroupBySection(clientValues, convertClientToServer);
         const serverValuesForMainValues = getServerValuesForMainValues(
             dataEntryFieldValues,
             dataEntryFieldsMeta,
             formFoundation,
         );
-
-        // $FlowFixMe
-        const attributes = Object.keys(serverValuesForFormValues)
-            .map(key => ({
-                attribute: key,
-                value: serverValuesForFormValues[key],
-            }));
+        const { enrolledAt, occurredAt } = serverValuesForMainValues;
+
+        const { stages } = getTrackerProgramThrowIfNotFound(programId);
+
+        const attributeCategoryOptionsId = 'attributeCategoryOptions';
+        const attributeCategoryOptions = Object.keys(serverValuesForMainValues)
+            .filter(key => key.startsWith(attributeCategoryOptionsId))
+            .reduce((acc, key) => {
+                const categoryId = key.split('-')[1];
+                acc[categoryId] = serverValuesForMainValues[key];
+                return acc;
+            }, {});
+
+        const formServerValues = serverValuesForFormValues[Section.groups.ENROLLMENT];
```

Comment:
Hey,
I am not able to save the TEI attributes. I dug a bit at it looks like `serverValuesForFormValues[Section.groups.ENROLLMENT]` is `undefined`. Can you have a look?

https://github.com/dhis2/capture-app/assets/6817073/c597c80d-e216-4228-a9a5-ca2c9b026afb


Thanks!